app/react/kubernetes/applications/components/RedeployGitAppForm/RedeployGitAppForm.tsx

```diff
@@ -0,0 +1,362 @@
+import { useRef, useState } from 'react';
+import { RefreshCw } from 'lucide-react';
+import { useRouter } from '@uirouter/react';
+
+import { Stack, StackId } from '@/react/common/stacks/types';
+import {
+  parseAutoUpdateResponse,
+  transformAutoUpdateViewModel,
+} from '@/react/portainer/gitops/AutoUpdateFieldset/utils';
+import { AutoUpdateFieldset } from '@/react/portainer/gitops/AutoUpdateFieldset';
+import {
+  AutoUpdateMechanism,
+  AutoUpdateModel,
+} from '@/react/portainer/gitops/types';
+import {
+  baseStackWebhookUrl,
+  createWebhookId,
+} from '@/portainer/helpers/webhookHelper';
+import { TimeWindowDisplay } from '@/react/portainer/gitops/TimeWindowDisplay';
+import { RefField } from '@/react/portainer/gitops/RefField';
+import { parseAuthResponse } from '@/react/portainer/gitops/AuthFieldset/utils';
+import { confirmEnableTLSVerify } from '@/react/portainer/gitops/utils';
+import { AuthFieldset } from '@/react/portainer/gitops/AuthFieldset';
+import { useGitCredentials } from '@/react/portainer/account/git-credentials/git-credentials.service';
+import { useCurrentUser } from '@/react/hooks/useUser';
+import { useAnalytics } from '@/react/hooks/useAnalytics';
+import { RepositoryMechanismTypes } from '@/kubernetes/models/deploy';
+import { useCreateGitCredentialMutation } from '@/react/portainer/account/git-credentials/queries/useCreateGitCredentialsMutation';
+import { useEnvironmentId } from '@/react/hooks/useEnvironmentId';
+import { notifySuccess } from '@/portainer/services/notifications';
+import { useStackQuery } from '@/react/common/stacks/queries/useStackQuery';
+
+import { FormSection } from '@@/form-components/FormSection';
+import { SwitchField } from '@@/form-components/SwitchField';
+import { LoadingButton } from '@@/buttons';
+import { buildConfirmButton } from '@@/modals/utils';
+import { confirm } from '@@/modals/confirm';
+import { ModalType } from '@@/modals';
+import { InlineLoader } from '@@/InlineLoader';
+import { Alert } from '@@/Alert';
+
+import { useUpdateKubeGitStackMutation } from './queries/useUpdateKubeGitStackMutation';
+import {
+  KubeAppGitFormValues,
+  RedeployGitStackPayload,
+  UpdateKubeGitStackPayload,
+} from './types';
+import { redeployGitAppFormValidationSchema } from './redeployGitAppFormValidationSchema';
+import { useRedeployKubeGitStackMutation } from './queries/useRedeployKubeGitStackMutation';
+
+type Props = {
+  stackId: StackId;
+  namespaceName: string;
+};
+
+export function RedeployGitAppForm({ stackId, namespaceName }: Props) {
+  const { data: stack, ...stackQuery } = useStackQuery(stackId);
+  const { trackEvent } = useAnalytics();
+  const initialValues = getInitialValues(stack);
+  const { user } = useCurrentUser();
+  const { data: gitCredentials, ...gitCredentialsQuery } = useGitCredentials(
+    user.Id
+  );
+  const environmentId = useEnvironmentId();
+  const createGitCredentialMutation = useCreateGitCredentialMutation();
```

Comment:
since we always create git credentials if needed for git stack, can I suggest doing it inside the mutation function? like here https://github.com/portainer/portainer/blob/dabcf4f7db042316dfa1dd9ab01f2640d870671d/app/react/portainer/templates/custom-templates/queries/useCreateTemplateMutation.ts#L83

Reply:
Sure, updated